packages/renderer/src/lib/image/ImageDetailsFiles.svelte

```diff
@@ -0,0 +1,69 @@
+<script lang="ts">
+import type { ImageLayerFile, ImageLayer } from '../../../../main/src/plugin/image-layers';
+import type { ImageInfoUI } from './ImageInfoUI';
+import { onMount } from 'svelte';
+import ImageFileTreeView from './ImageFileTreeView.svelte';
+import { ImageUtils } from './image-utils';
+import Spinner from '../ui/Spinner.svelte';
+import type { FileNode } from '../../../../main/src/plugin/file-tree';
+
+let currentRoot: FileNode<ImageLayerFile> | undefined; // keep this variable declaration first (for the tests to access it at $$.ctx[1] - layers, its dependency, is at 0)
+
+export let image: ImageInfoUI;
+let loading = true;
+let layers: ImageLayer[];
+let currentLayerId: string;
+$: currentRoot = layers?.length ? layers[0].tree.root : undefined;
+
+onMount(async () => {
+  layers = await window.getImageLayers(image.engineId, image.id);
```

Comment:
I have an image, which take forever on the screen `Loading layers, please wait...`, so I quite the page, come back, and do that 2-3 times, when looking at the code I understand that each time I opened the container tab files I re-start a full extraction. 

I think we should have either inside the preload or the container registry a place where we know which one are still running, as some may take very long time (1GB image took nearly 3 minutes)

In the future we might want also to cache this information

Reply:
Yes, there are many things I would like to add after this first version is in. For example, I would like to show from the beginning the different layers (we can get them very quickly), then show which layer is being "analyzed" (this taks time), so the user has an idea of how many time it will need to wait